File: Sakila_Php/Entities/FilmActor.php
<?PHP
class FilmActor implements JsonSerializable{

   private $lastUpdate;


   public function getActorId(){
        return $this->actorId;
   }
   public function setActorId($actorId){
        $this->actorId = $actorId;
   }
   
   public function getFilmId(){
        return $this->filmId;
   }
   public function setFilmId($filmId){
        $this->filmId = $filmId;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function FilmActor($actorId,$filmId,$lastUpdate=''){
		   $this->setActorId($actorId);
		   $this->setFilmId($filmId);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($actorId,$filmId,$lastUpdate='') {
		   $this->setActorId($actorId);
		   $this->setFilmId($filmId);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "actorId" . chr(34) . ":" . chr(34) . $this->getActorId() . chr(34) 
		    . "," . chr(34) . "filmId" . chr(34) . ":" . chr(34) . $this->getFilmId() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new FilmActor();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>

File: Sakila_Net/Sakila_Net/VB_SakilaLib/Model/FilmActor.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClFilmActor

   private  actorId  as ClActor
   private  filmId  as ClFilm
   private lastUpdate as Date


   public Property DataActorId()  as ClActor
        Get
            Return actorId
        End Get
        Set(ByVal Value  as ClActor)
            actorId = Value
        End Set        
   End Property   
   public Property DataFilmId()  as ClFilm
        Get
            Return filmId
        End Get
        Set(ByVal Value  as ClFilm)
            filmId = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (actorId  as ClActor,filmId  as ClFilm,lastUpdate    as Date)
		   dataActorId = actorId
		   dataFilmId = filmId
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (actorId  as ClActor,filmId  as ClFilm)
    dataActorId = actorId
    dataFilmId = filmId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "actorId" & chr(34) & ":" & chr(34) & dataActorId.ToString() & chr(34) _
		    & "," & chr(34) & "filmId" & chr(34) & ":" & chr(34) & dataFilmId.ToString() & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  
